### PR TITLE
[Merged by Bors] - feat: Extend `rfl` tactic to use `refl` attribute.

### PR DESCRIPTION
**Pull request by newell**
This extends the `rfl` tactic in Lean 4 to use `refl` attribute.

### CODE REVIEW

File: Mathlib/Tactic/Relation/Rfl.lean [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/-- look up reflexivity lemmas -/
+def reflLemmas (env : Environment) : DiscrTree Name :=
+  reflExtension.getState env
+
+open Lean.Elab.Tactic in
+
+/-- This tactic applies to a goal whose target has the form `x ~ x`, where `~` is a reflexive
+relation, that is, a relation which has a reflexive lemma tagged with the attribute [refl].
+-/
+elab "rfl" : tactic =>
```
Comment (digama0) [inline Mathlib/Tactic/Relation/Rfl.lean:50]: Ah, this is the problem. Using `elab "rfl"` means you are declaring another syntax called `rfl`, which takes priority over the other one and all its implementations. You want to use ``elab_rules : tactic | `(rfl) => ...`` instead here to make your implementation play well with the others.

File: Mathlib/Tactic/Relation/Rfl.lean [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+elab_rules : tactic
+| `(tactic| rfl) =>
+  withMainContext do
+  let tgt ← getMainTarget
+  match ← relationAppM? tgt with
+  | some (rel, _, _) =>
+    let s ← saveState
+    for lem in ← (reflLemmas (← getEnv)).getMatch rel do
+      try
+        liftMetaTactic (MVarId.apply · (← mkConstWithFreshMVarLevels lem))
```
Comment (semorrison) [inline Mathlib/Tactic/Relation/Rfl.lean:65]: You can even write here:
```suggestion
        liftMetaTactic (·.apply (← mkConstWithFreshMVarLevels lem))
```

File: Mathlib/Mathport/Syntax.lean [hunk 3]
```diff
@@ -27,6 +27,7 @@ import Mathlib.Tactic.NormNum
 import Mathlib.Tactic.PushNeg
 import Mathlib.Tactic.Recover
 import Mathlib.Tactic.Replace
+import Mathlib.Tactic.Rfl
```
Comment (semorrison) [inline Mathlib/Mathport/Syntax.lean:30]:
```suggestion
import Mathlib.Tactic.Relation.Rfl
```

**Comment by newell** [comment]
Updated with review fixes.  This branch relies on #425 and that was done in a separate branch since that code was from @siddhartha-gadgil.

**Comment by digama0** [comment]
bors merge

**Comment by digama0** [comment]
bors merge

**Comment by digama0** [comment]
bors merge

**Pull request closed without merging**